docs/api/js/utils.md

```diff
@@ -1,68 +1,772 @@
+        .map((log) => {
+            const sender = `0x${log.topics[1].slice(log.topics[1].length - addressBytesLen)}`;
+            const bytesCodehash = log.topics[2];
+            const address = `0x${log.topics[3].slice(log.topics[3].length - addressBytesLen)}`;
+            return {
+                sender: utils.getAddress(sender),
+                bytecodeHash: bytesCodehash,
+                deployedAddress: utils.getAddress(address)
+            };
+        });
+
+    return deployedContracts;
+}
+```
+
+### `getERC20BridgeCalldata`
+
+Returns the calldata sent by an L1 ERC20 bridge to its L2 counterpart.
 
-export const RECOMMENDED_GAS_LIMIT = {
-    DEPOSIT: 600_000,
-    EXECUTE: 620_000,
-    ERC20_APPROVE: 50_000
-};
+#### Inputs
+
+- `l1TokenAddress`: token address on L1 as string.
+- `l1Sender`: sender address on L1 as string.
+- `l2Receiver`: recipient address on L2 as string.
+- `amount`: gas fee for the number of tokens to bridge as `BigNumberish` object.
+- `provider`: ethers `Provider` object.
+
+```ts
+export async function getERC20BridgeCalldata(
+    l1TokenAddress: string,
+    l1Sender: string,
+    l2Receiver: string,
+    amount: BigNumberish,
+    provider: ethers.providers.Provider
+): Promise<string> {
+    const gettersData = await getERC20GettersData(l1TokenAddress, provider);
+    return L2_BRIDGE_ABI.encodeFunctionData('finalizeDeposit', [
+        l1Sender,
+        l2Receiver,
+        l1TokenAddress,
+        amount,
+        gettersData
+    ]);
+}
+```
+
+### `getERC20GettersData`
+
+Gets the data needed for initializing an L1 token counterpart on L2.
+
+#### Inputs
+
+- `l1TokenAddress`: token address on L1 as string.
+- `provider`: ethers `Provider` object.
+
+#### Outputs
+
+An ABI-encoded array of:
+
+- `nameBytes`: `bytes` object representation of token name.
+- `symbolBytes`: `bytes` object representation of token symbol.
+- `decimalBytes`: `bytes` object representation of token decimal representation.
+
+```ts
+async function getERC20GettersData(l1TokenAddress: string, provider: ethers.providers.Provider): Promise<string> {
+    const token = IERC20MetadataFactory.connect(l1TokenAddress, provider);
+
+    const name = await token.name();
+    const symbol = await token.symbol();
+    const decimals = await token.decimals();
+
+    const coder = new AbiCoder();
+
+    const nameBytes = coder.encode(['string'], [name]);
+    const symbolBytes = coder.encode(['string'], [symbol]);
+    const decimalsBytes = coder.encode(['uint256'], [decimals]);
+
+    return coder.encode(['bytes', 'bytes', 'bytes'], [nameBytes, symbolBytes, decimalsBytes]);
+}
+```
+
+### `getL2HashFromPriorityOp`
+
+Returns the hash of the L2 priority operation from a given transaction receipt and L2 address.
+
+#### Inputs
+
+- `txReceipt`: receipt of the L1 transaction as ethers `TransactionReceipt` object.
+- `zkSyncAddress`: address of sender/transaction ?? on L2 as `Address` object.
```

Comment:
this one is the address of the zkSync main contract on L1

Reply:
done